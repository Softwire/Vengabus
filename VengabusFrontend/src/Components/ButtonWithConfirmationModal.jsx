import React from 'react';
import { Modal, Alert, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithConfirmationModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleOpen = () => {
        const afterOpen = (this.props.afterShowModalAction || (() => { }));
        this.setState({ show: true }, afterOpen);
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        const afterClose = (this.props.afterCloseModalAction || (() => { }));
        this.setState(defaultState, afterClose);
    }

    render() {
        const style = this.props.buttonDisabled ? {
            pointerEvents: 'none'
        } : {};

        const tooltip = (
            <Tooltip id="tooltip">
                <strong>{this.props.tooltipMessage}</strong>
            </Tooltip>
        );

        const button =
            <Button
                onClick={this.handleOpen}
                bsStyle={this.props.buttonStyle || "danger"}
                disabled={!!this.props.buttonDisabled}
                style={style}
            >
                {this.props.buttonText}
            </Button>;

        const buttonWithToolTip =
            <OverlayTrigger rootClose overlay={tooltip} placement="top">
                <div style={{ display: 'inline-block', cursor: 'not-allowed', float: 'right' }}>
                    {button}
                </div>
            </OverlayTrigger>;


        return (
            <React.Fragment>
                {this.props.tooltipMessage ? buttonWithToolTip : button}
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {this.props.modalBody ?
                            <Alert bsStyle={this.props.buttonStyle ? this.props.buttonStyle : "danger"}>
                                {this.props.modalBody}
                            </Alert > : null
                        }
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText || "Cancel"}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment >
        );
    }
}

export { ButtonWithConfirmationModal };
