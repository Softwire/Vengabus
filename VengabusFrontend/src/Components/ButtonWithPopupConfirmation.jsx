import React from 'react';
import { Modal, Alert, Button } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithPopupConfirmation extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        this.props.closeModalAction();
        this.setState(defaultState);
    }

    handleOpening = () => {
        this.setState({ show: true });
        this.props.showModalAction();
    }

    render() {
        return (
            <React.Fragment>
                <Button onClick={this.handleOpening} bsStyle={this.props.buttonStyle}>
                    {this.props.buttonText}
                </Button>
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <Alert bsStyle="danger">
                            {this.props.modalBody}
                        </Alert >
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment>
        );
    }
}

export { ButtonWithPopupConfirmation };
