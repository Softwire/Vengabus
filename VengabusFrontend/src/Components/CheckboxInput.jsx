import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 * @prop {string} text The name of the property that is being set by the checkbox.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <Checkbox
                    className={leftAlign}
                    checked={this.props.data}
                    onChange={(event) => {
                        this.props.onChange(event.target.checked);
                    }}
                />
            </div>
        );
    }

}


