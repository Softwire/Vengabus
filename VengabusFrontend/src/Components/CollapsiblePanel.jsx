import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            glyphicon: "plus"
        };
    }


    toggleGlyphicon = () => {
        this.setState({
            glyphicon: this.state.glyphicon === "plus" ? "minus" : "plus"
        });
    }

    render() {
        const panelStyle = css`
            .panel {
                margin-bottom: 10px;
            }
            .panel-heading {
                padding: 0 15px;
                color: #333;
                background-color: #f5f5f5;
                border-color: #ddd;
            }
            .panel-body { 
                padding: 0;
            }
            pre {
                background-color: #ffffff;
                border: none;
                margin: 0;
            }
            pre p {
                margin: 0;
            }
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const gylphiconStyle = css`
            margin-right: 10px;
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={panelStyle} onToggle={() => this.toggleGlyphicon()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title >
                            <Glyphicon glyph={this.state.glyphicon} className={gylphiconStyle} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.panelContent}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}