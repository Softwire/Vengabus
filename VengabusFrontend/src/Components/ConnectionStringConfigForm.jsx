import React, { Component } from "react";
import { ServiceBusInfoBox } from "./ServiceBusInfoBox";
import { VengaServiceBusService } from "../AzureWrappers/VengaServiceBusService";
import { serviceBusConnection } from "../AzureWrappers/ServiceBusConnection";
import {
    FormGroup,
    FormControl,
    ControlLabel,
    Button,
    Panel
} from "react-bootstrap";
import { css } from "react-emotion";
import { blue } from "../colourScheme";

export const LOCAL_STORAGE_STRINGS = Object.freeze({
    ConnectionString: "connectionString",
    APIroot: "apiRoot"
});

/**
 *    Local Storage is accessible as a global window when run in the browser.
 *    This variable exists mostly to be a holder for this comment! The global
 *    variable will be stubbed in by the jest-localstorage-mock package, during testing.
 */
const localStorageAccessor = localStorage;

export class ConnectionStringConfigForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            connStringVal: "",
            APIroot: "",
            info: ""
        };
    }

    componentWillMount() {
        const localStorageConnectionString =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.ConnectionString) ||
            "";
        this.updateConString(localStorageConnectionString);
        const localStorageApiRoot =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.APIroot) ||
            "";
        this.updateAPIroot(localStorageApiRoot);
    }

    /** Updates the value of the connection string in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newConString The updated value of the connection string.  
     */
    updateConString = newConString => {
        this.setState({ connStringVal: newConString });
        serviceBusConnection.setConnectionString(newConString);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.ConnectionString,
            newConString
        );
    };

    /** Updates the value of the API root location in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newURI The updated value of the API root.  
     */
    updateAPIroot = newURI => {
        Math.random();
        this.setState({ APIroot: newURI });
        serviceBusConnection.setAPIroot(newURI);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.APIroot,
            newURI
        );
    }

    // Called whenever the value of the connection string input box changes.
    handleConnectionChange = event => {
        this.updateConString(event.target.value);
    };

    // Called whenever the value of the API server input box changes.
    handleAPIChange = event => {
        this.updateAPIroot(event.target.value);
    }

    /**
     * Updates the info in the sidebar based on the current status of VengaServiceBusService.
     * Called whenver the connect button is pressed.
     */
    submitConnectionStringClick = () => {
        const infoPromise = VengaServiceBusService.getServiceBusProperties(
            this.state.connStringVal
        );
        infoPromise
            .then(response => {
                this.setState({
                    info: response
                });
            })
            .catch(error => {
                console.log(error);
            });
    };

    render() {
        const inputStyle = css`
      color: black;
    `;

        const buttonStyle = css`
      color: black;
      margin: 5px;
    `;

        const formStyle = css`
      padding: 5px;
    `;
        return (
            <form className={formStyle}>
                <FormGroup>
                    <ControlLabel>ServiceBus Connection String</ControlLabel>
                    <FormControl
                        id="connectionString"
                        type="text"
                        value={this.state.connStringVal}
                        placeholder="Enter Connection String"
                        onChange={this.handleConnectionChange}
                    />
                    <FormControl.Feedback />

                    <ControlLabel>ServiceBus API Server String</ControlLabel>
                    <FormControl
                        id="APILocationForm"
                        type="text"
                        value={this.state.APIroot}
                        placeholder="Enter API Server Location"
                        onChange={this.handleAPIChange}
                    />
                    <FormControl.Feedback />

                </FormGroup>
                <Button
                    className={buttonStyle}
                    onClick={this.submitConnectionStringClick}
                >
                    Connect
                </Button>
                {
                    //buttons want to grip on to the top of things not pretty so add a break to separate
                }
                <div>
                    <br />
                </div>
                <ServiceBusInfoBox
                    connStringVal={this.state.connStringVal}
                    info={this.state.info}
                />
            </form>
        );
    }
}
