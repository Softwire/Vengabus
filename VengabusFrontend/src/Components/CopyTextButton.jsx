import React from 'react';
import { Button, Overlay, Tooltip } from 'react-bootstrap';
import ReactDOM from 'react-dom';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import { css } from 'emotion';

/**
 * Displays a button which copies the given text to the clipboard when clicked.
 * @prop {string} text The text to copy.
 */
export class CopyTextButton extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.text = props.text;

        this.state = {
            show: false
        };
    }

    componentWillUnmount() {
        if (this.hideCopiedTooltip) {
            clearTimeout(this.hideCopiedTooltip);
        }
    }

    /**
     * Returns the component which the tooltip is pointing to.
     * @return {object} The component.
     */
    getTarget = () => {
        return ReactDOM.findDOMNode(this.target);
    }

    /**
     * Shows a 'copied' tooltip when the button is pressed.
     */
    handleToggle = () => {
        this.setState({ show: true });
        this.hideCopiedTooltip = setTimeout(() => {
            this.setState({ show: false });
        }, 2000);
    }

    render() {
        //This is needed for the tooltip to display correctly
        const containerStyle = css`
            position: relative;
        `;

        return (
            <CopyToClipboard text={this.text}>
                <div className={containerStyle}>
                    <Button
                        ref={button => {
                            this.target = button;
                        }}
                        onClick={this.handleToggle}
                    >
                        Copy Body
                    </Button>

                    <Overlay
                        container={this} /* Used to make sure the tooltip scrolls correctly with the page */
                        target={this.getTarget} /* The tooltip should appear above the copy button */
                        show={this.state.show} /* The tooltip should only be shown when this.state.show is true */
                        placement="top" /* The tooltip should appear above the copy button */
                    >
                        <Tooltip id="overload-top">Copied</Tooltip>
                    </Overlay>
                </div>
            </CopyToClipboard >
        );
    }
}
