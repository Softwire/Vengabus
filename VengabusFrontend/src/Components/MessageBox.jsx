import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';
import { FormattingBox } from './FormattingBox';
import { CopyTextButton } from './CopyTextButton';
import { css } from 'emotion';
import { CollapsiblePanel } from './CollapsiblePanel';

export class MessageBox extends Component {

    getJSXFromProperties = (propertiesArray) => {
        let propertiesJSX = [];
        for (let propName in propertiesArray) {
            let currentPropValue = propertiesArray[propName];
            if (propName !== 'messageId' && currentPropValue !== null) {
                propertiesJSX.push(
                    <p key={propName}>{propName + ': ' + String(currentPropValue)}</p> /*string conversion needed to display booleans properly*/
                );
            }
        }
        return propertiesJSX.length > 0 ? propertiesJSX : null;
    }

    render() {
        const buttonStyle = css`
            float: right;
            margin-left: 5px;
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const messageIdStyle = css`
            display: inline-block;
        `;
        const message = this.props.message;
        if (!message) {
            return null;
        }
        const preDefinedPropsJSX = this.getJSXFromProperties(message.predefinedProperties) || <p>There are no pre-defined properties to display</p>;
        const customPropsJSX = this.getJSXFromProperties(message.customProperties) || <p>There are no user-defined properties to display</p>;
        return (
            <div className="static-modal" >
                <Modal show={this.props.show} onHide={this.props.handleClose} >
                    <Modal.Header>
                        <Modal.Title className={headerStyle}>Message Id:&nbsp;</Modal.Title>
                        <Modal.Title className={messageIdStyle}>{message.predefinedProperties.messageId}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <CollapsiblePanel panelTitle={"Pre-defined Properties"} panelContent={<pre>{preDefinedPropsJSX}</pre>}  />
                        <CollapsiblePanel panelTitle={"User-defined Properties"} panelContent={<pre>{customPropsJSX}</pre>} />
                        <FormattingBox
                            data={message.messageBody}
                        />
                    </Modal.Body>

                    <Modal.Footer>
                        <Button className={buttonStyle} onClick={this.props.handleClose}>Close</Button>
                        <CopyTextButton className={buttonStyle} text={this.props.messageBody} />
                    </Modal.Footer>
                </Modal>
            </div>

        );
    }
}