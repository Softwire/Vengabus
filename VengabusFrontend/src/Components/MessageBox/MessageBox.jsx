import React, { Component } from 'react';
import { Modal, Button, ButtonToolbar } from 'react-bootstrap';
import { CopyTextButton } from '../CopyTextButton';
import { css } from 'emotion';
import { CollapsiblePanel } from '../CollapsiblePanel';
import { PAGES, pageSwitcher } from '../../Pages/PageSwitcherService';
import { FormattingBox } from './FormattingBox';

export class MessageBox extends Component {

    convertMessagePropertiesToJSXArray = (propertiesArray) => {
        const propertiesJSX = [];
        for (let propName in propertiesArray) {
            let currentPropValue = propertiesArray[propName];
            if (propName !== 'messageId') {
                propertiesJSX.push(
                    <p key={propName}>{propName + ': ' + String(currentPropValue)}</p> /*string conversion needed to display booleans properly*/
                );
            }
        }
        return propertiesJSX.length > 0 ? propertiesJSX : null;
    }

    handleReplayMessage = (message) => {
        //qq change hardcoded recipientIsQueue and selectedQueue later
        pageSwitcher.switchToPage(
            PAGES.SendMessagePage,
            { message: message, recipientIsQueue: true, selectedQueue: 'demoqueue1' }
        );
    }

    render() {
        const buttonToolbarStyle = css`
            margin-right: -8px; /* This counteracts the right margin for the "first" (right-most) button. See comment below. */
            
            &.btn-toolbar .btn {
                margin-right: 8px; /* The tooltip library's layout code doesn't handle margins, so we have to have the margin on the right. */
                margin-left: 0px;
                float: right;
            }
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const messageIdStyle = css`
            display: inline-block;
        `;
        const panelStyle = css`
            .panel {
                margin-bottom: 15px;
            }
        `;
        const message = this.props.message;
        if (!message) {
            return null;
        }
        const preDefinedPropsJSX = this.convertMessagePropertiesToJSXArray(message.predefinedProperties) || <p>There are no pre-defined properties to display</p>;
        const customPropsJSX = this.convertMessagePropertiesToJSXArray(message.customProperties) || <p>There are no user-defined properties to display</p>;
        return (
            <div className="static-modal" >
                <Modal show={this.props.show} onHide={this.props.handleClose} id="messageBoxModal" >
                    <Modal.Header>
                        <Modal.Title className={headerStyle}>Message Id:&nbsp;</Modal.Title>
                        <Modal.Title className={messageIdStyle}>{message.predefinedProperties.messageId}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body className={panelStyle} id="messageBoxModalBody">
                        <CollapsiblePanel panelTitle={"Pre-defined Properties"}>
                            <pre>{preDefinedPropsJSX}</pre>
                        </CollapsiblePanel>
                        <CollapsiblePanel panelTitle={"User-defined Properties"}>
                            <pre>{customPropsJSX}</pre>
                        </CollapsiblePanel>
                        <FormattingBox message={this.props.message.messageBody} />
                    </Modal.Body>

                    <Modal.Footer>
                        <ButtonToolbar className={buttonToolbarStyle}>
                            { /*Note that these buttons are rendered in order, Right-to-Left*/}
                            <Button onClick={this.props.handleClose} id="messageBoxClose">Close</Button>
                            <CopyTextButton text={message.messageBody} id="messageBoxCopy" />
                            <Button onClick={() => this.handleReplayMessage(message)} id="messageBoxReplayMessage">Replay Message to demoqueue1</ Button>
                        </ButtonToolbar>
                    </Modal.Footer>
                </Modal>
            </div>

        );
    }
}