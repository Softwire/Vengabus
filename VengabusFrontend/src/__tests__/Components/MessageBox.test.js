import { MessageBox } from '../../Components/MessageBox';
import renderer from 'react-test-renderer';
import React from 'react';

describe('MessageBox', () => {

    const showMessage = true;
    const message = { predefinedProperties: [{ messageId: 'id1' }], messageBody: "Hello world!", customProperties: [{ userDefinedProp1: "value 1" }, { userDefinedProp2: "value 2" }] };

    it('renders correctly with given props', () => {
        let messageBox = renderer.create(
            <MessageBox
                message={message}
            />);
        expect(messageBox.toJSON()).toMatchSnapshot();
    });

});
