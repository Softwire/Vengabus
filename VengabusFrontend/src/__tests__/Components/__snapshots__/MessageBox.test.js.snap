// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders correctly with given props 1`] = `
<div
  className="static-modal"
/>
`;
